### DISCUSSION TITLE
NextJS returns HTML error pages for non page assets like SVG, CSS, etc.

### DISCUSSION THREAD

**Discussion by jaworek** [vercel/next.js#24461] [Help]
Hi!

I'm working on a big project that uses NextJS (10.1) with custom server. Recently we had a problem with server overload due to some issues that caused many assets to return 404 and 500 errors, this in turn made a big number of API requests that crashed the server. Each one of them was an HTML error page, where it seems only an error status number would be sufficient. My initial idea was that there is some misconfiguration in our custom server, but when I created a fresh project, I see the same behaviour.
Is it expected? Are there any ways to mitigate this issue or make it perform better?
Maybe I'm looking at the wrong place and the issue is somewhere else?

**Comment by apollonian** [comment]
Could you add the reproducible example or a CodeSandbox? That would be very helpful, thanks!

**Reply by jaworek** [reply]
Reproduction:
1. Create new NextJS project.
2. Remove one of the .svg files.
3. Run `yarn dev`
4. Open Network tab in DevTools
5. Look for the request for this .svg file
6. It should be a 404 response. When you click on the preview you will see 404 HTML page which seems unnecessary in this case.
